service/arm_templates/node_setup.json

```diff
@@ -179,22 +177,27 @@
             "version": "[parameters('osImageVersion')]"
           },
           "osDisk": {
-            "name": "[variables('osDiskName')]",
-            "vhd": {
-              "uri": "[concat('http://', parameters('storageAccountName'), '.blob.core.windows.net/vhds/', variables('osDiskName'), '.vhd')]"
+            "name": "[concat(parameters('clusterID'), '-', parameters('prefix'), '-', parameters('osDisk').name)]",
+            "caching": "[parameters('osDisk').caching]",
+            "managedDisk": {
+              "storageAccountType": "[parameters('osDisk').storageAccountType]"
             },
-            "caching": "ReadWrite",
             "createOption": "FromImage"
           },
-          "dataDisks": [
+          "copy": [
             {
-              "name": "[variables('dataDiskName')]",
-              "diskSizeGB": "[parameters('dataDiskSizeGB')]",
-              "lun": 0,
-              "vhd": {
-                "uri": "[concat('http://', parameters('storageAccountName'), '.blob.core.windows.net/vhds/', variables('dataDiskName'), '.vhd')]"
-              },
-              "createOption": "Empty"
+              "name": "dataDisks",
+              "count": "[length(parameters('dataDisks'))]",
+              "input": {
+                "name": "[concat(parameters('clusterID'), '-', parameters('prefix'), '-', parameters('dataDisks')[copyIndex('dataDisks')].name)]",
+                "caching": "[parameters('dataDisks')[copyIndex('dataDisks')].caching]",
+                "diskSizeGB": "[parameters('dataDisks')[copyIndex('dataDisks')].sizeGB]",
+                "managedDisk": {
+                  "storageAccountType": "[parameters('dataDisks')[copyIndex('dataDisks')].storageAccountType]"
+                },
+                "createOption": "Empty",
+                "lun": "[add(copyIndex('dataDisks'), 1)]"
+              }
```

Comment:
I did my homework. We don't want to expose that in the custom object at all. Let's create disks with Standard_LRS type, 50gb for data disk. All fixed in the template. Maybe extract to variables if makes sense.

Ping @r7vme

Reply:
@kopiczko I had a chat with @r7vme last week and the solution seemed to add a docker disk on both the master and worker, as well as an etcd disk on the master.

If I hardcode that, the etcd disk will also be premium with a size of 50gb, which feels a bit overkill.

Should we rather merge the etcd and docker disk together? Or maybe there's a better solution?

Reply:
@r7vme I think we can live with HDD for docker and SSD for etcd. WDYT?

Let's create as many disks as we need. Separate disk for etcd is legit. As well as separate disk for docker and data. What I mean is just not exposing that in the custom object (azuretpr) because we don't need that. This gives us freedom to change that any time.

Reply:
Looks like we should also take into account VM size

> Premium Storage supports DS-series, DSv2-series, GS-series, Ls-series, and Fs-series VMs. You can use standard and premium storage disks with these VM types. You cannot use premium storage disks with VM series that are not Premium Storage-compatible.

We can just replicate what acs-engine does: They select storage based on VM size [2].

For example, if user requested `Standard_A3` we can only use Standard storage. But with `Standard_DS1_v2` we can use both, but i would stick with Premium always if it's possible.

1 -  https://docs.microsoft.com/en-us/azure/virtual-machines/windows/premium-storage#supported-vms
2 - https://github.com/Azure/acs-engine/blob/master/pkg/acsengine/azureconst.go#L525-L1049

Reply:
I'd go with the Premium always and don't care at this point. As I said we don't have that in the custom object. We don't have to worry.

Reply:
Good idea, I like that. @kopiczko I'll close the azuretpr since we don't need that anymore.

I'll keep you guys posted with updates to this PR.